airbyte-integrations/base/base.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+
+set -e
+
+function echo2() {
+  echo >&2 "$@"
+}
+
+function error() {
+  echo2 "$@"
+  exit 1
+}
+
+# todo: make it easy to select source or destination and validate based on selection
+function main() {
+  CMD="$1"
+  shift 1
+
+  ARGS=
+  while [ $# -ne 0 ]; do
+    case "$1" in
+    --config)
+      CONFIG_FILE="$2"
+      shift 2
+      ;;
+    --catalog)
+      CATALOG_FILE="$2"
+      shift 2
+      ;;
+    --state)
+      STATE_FILE="$2"
+      shift 2
+      ;;
+    *)
+      error "Unknown option: $1"
+      ;;
+    esac
+  done
+
+  case "$CMD" in
+  spec)
+    eval "$AIRBYTE_SPEC_CMD"
```

Comment:
Why do you need to `eval`?

Reply:
Excluding eval doesn't allow multiword commands to be passed in.
```
# jrhizor in ~/code/airbyte on git:jrhizor/better-py-exp ✖︎ [10:50:12]
→ CMD="echo hi"
# jrhizor in ~/code/airbyte on git:jrhizor/better-py-exp ✖︎ [10:53:06]
→ $CMD
zsh: command not found: echo hi
```

Eval seems the preferred option here compared to something like:
```
 cmd_arr=($(echo "$AIRBYTE_SPEC_CMD" | tr ' ' '\n'))
 "${cmd_arr[@]}"
```